src/cacheHttpClient.ts

```diff
@@ -109,8 +185,8 @@ export async function getCacheEntry(
         keys.join(",")
     )}&version=${version}`;
 
-    const response = await httpClient.getJson<ArtifactCacheEntry>(
-        getCacheApiUrl(resource)
+    const response = await retryTypedResponse("getCacheEntry", () =>
+        httpClient.getJson<ArtifactCacheEntry>(getCacheApiUrl(resource))
```

Comment:
The http-client has built-in retries for read operations that we could use for this call and `downloadCache`. It's on the `IRequestOptions` and is ignored for write operations.

Reply:
I'd have to check again, but I think one issue with http-client is that it only retries on certain http status codes.  If there's a socket-level error, it wouldn't retry.  But as you say, this is certainly something that could be put into http-client.